src/HHbbVV/processors/bbVVSkimmer.py

```diff
@@ -684,6 +692,274 @@ def process(self, events: ak.Array):
+        vbfJets_sorted_pt = ak.pad_none(
+            vbfJets_sorted_pt, 2, clip=True
+        )  # this is the only which does not guarantee two guys. in the other sorts, the entries are specifically None.
+
+        # pt sorted eta and dijet mass mask
+        vbf1 = vector.array(
+            {
+                "pt": ak.flatten(pad_val(vbfJets_sorted_pt[:, 0:1].pt, 1, axis=1)),
+                "phi": ak.flatten(pad_val(vbfJets_sorted_pt[:, 0:1].phi, 1, axis=1)),
+                "eta": ak.flatten(pad_val(vbfJets_sorted_pt[:, 0:1].eta, 1, axis=1)),
+                "M": ak.flatten(pad_val(vbfJets_sorted_pt[:, 0:1].mass, 1, axis=1)),
+            }
+        )
+
+        vbf2 = vector.array(
+            {
+                "pt": ak.flatten(pad_val(vbfJets_sorted_pt[:, 1:2].pt, 1, axis=1)),
+                "phi": ak.flatten(pad_val(vbfJets_sorted_pt[:, 1:2].phi, 1, axis=1)),
+                "eta": ak.flatten(pad_val(vbfJets_sorted_pt[:, 1:2].eta, 1, axis=1)),
+                "M": ak.flatten(pad_val(vbfJets_sorted_pt[:, 1:2].mass, 1, axis=1)),
+            }
+        )
+
+        jj = vbf1 + vbf2
+
+        mass_jj_cut_sorted_pt = jj.mass > 500
+        eta_jj_cut_sorted_pt = np.abs(vbf1.eta - vbf2.eta) > 4.0
+
+        vbfJets_mask_sorted_pt = vbfJets_mask  # * mass_jj_cut_sorted_pt * eta_jj_cut_sorted_pt # uncomment these last two to include dijet cuts
+        n_good_vbf_jets_sorted_pt = ak.fill_none(ak.sum(vbfJets_mask_sorted_pt, axis=1), 0)
+
+        # add vbf gen quark info
+        if isGen:  # add | True when debugging with local files
+            vbfGenJets = events.GenPart[events.GenPart.hasFlags(["isHardProcess"])][:, 4:6]
+
+            vbfVars[f"vbfptGen"] = pad_val(vbfGenJets.pt, 2, axis=1)
+            vbfVars[f"vbfetaGen"] = pad_val(vbfGenJets.eta, 2, axis=1)
+            vbfVars[f"vbfphiGen"] = pad_val(vbfGenJets.phi, 2, axis=1)
+            vbfVars[f"vbfMGen"] = pad_val(vbfGenJets.mass, 2, axis=1)
+
+            jet_pairs = ak.cartesian({"reco": vbfJets_sorted_pt[:, 0:2], "gen": vbfGenJets[:, 0:2]})
+
+            # Calculate delta eta and delta phi for each pair
+            delta_eta = jet_pairs["reco"].eta - jet_pairs["gen"].eta
+            delta_phi = np.pi - np.abs(np.abs(jet_pairs["reco"].phi - jet_pairs["gen"].phi) - np.pi)
+
+            # Calculate delta R for each pair
+            delta_R = np.sqrt(delta_eta**2 + delta_phi**2)
+
+            # Apply a mask for a low delta R value
+            mask_low_delta_R = delta_R < 0.4
+            num_per_event = ak.sum(mask_low_delta_R, axis=-1)  # miscounts 0's since some are empty
+
+            # Combine masks with logical 'and' operation
+            total_mask = n_good_vbf_jets_sorted_pt > 1
+
+            # set event that fail to have 0 for num of events.
+            num_per_event = ak.where(total_mask, num_per_event, 0)
+            vbfVars[f"vbfNumMatchedGen"] = num_per_event.to_numpy()
+
+            # adds data about R1 R2 selection efficiencies.
```

Comment:
Oh, just noticed this - you said you'll move this scan into the post-processing right?

Reply:
Or I was thinking of keeping this mask and "# int list representing the number of passing vbf jets per event
        vbfVars[f"nGoodVBFJets"] = n_good_vbf_jets_sorted_pt.to_numpy()" mask in the skimmer since this matched gen filter is only applied to vbf signal files and so it shouldn't be too small and the nGoodVBFJets mask uses some information about the ak4 jets that I don't keep (puId and Jet Id). I am planning to create masks in postprocessing based on dijet mass cut and dijet eta cut. I can also investigate R1 and R2. 
        
        If I want to optimize the script for this exact job submission, then I won't actually need any of the gen info and so I'll actually remove this for now

Reply:
To study R1 and R2 it seems like the HVV parT output will have a very large impact but I think the particle net HVV score should have similar effects